runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -982,6 +982,9 @@ private void pruneRemovedRouters(@Nonnull Layout layout) {
                     } catch (InterruptedException ie) {
                         throw new UnrecoverableCorfuInterruptedError(
                                 "Interrupted during layout fetch", ie);
+                    } catch (WrongClusterException we) {
+                        // It is futile trying to re-connect to the wrong cluster
+                        throw new UnrecoverableCorfuError("WrongClusterException:", we);
```

Comment:
Doesn't `AbstractView` have the same issue ?

Reply:
thanks fixed.